src/app/components/newproposal/RatioInput.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useEffect } from 'react';
+
+import { Box, Flex } from 'rebass/styled-components';
+import styled from 'styled-components';
+
+import { PROPOSAL_CONFIG } from 'app/containers/NewProposalPage/constant';
+import { Typography } from 'app/theme';
+import { useProposalType } from 'store/proposal/hooks';
+
+import Tooltip from '../Tooltip';
+
+export interface Recipient {
+  name?: string;
+  percent: string;
+}
+
+export interface RatioValue {
+  name?: string;
+  percent: string;
+}
+interface RatioProps {
+  onRatioChange: (value, name) => void;
+  showErrorMessage?: boolean;
+  value?: { [key: string]: string };
+  message?: string;
+}
+export default function RatioInput({ onRatioChange, showErrorMessage = false, value, message }: RatioProps) {
+  const [ratioValues, setRatioValues] = React.useState<Array<RatioValue> | undefined>();
+  const [tooltipPosition, setTooltipPosition] = React.useState('');
+
+  const selectedProposalType = useProposalType();
+
+  useEffect(() => {
+    if (selectedProposalType !== 'Text') {
+      (async () => {
+        const result = await PROPOSAL_CONFIG[selectedProposalType].fetchInputData();
+
+        setRatioValues(result);
+      })();
+    } else {
+      setRatioValues([]);
+    }
+  }, [selectedProposalType]);
+
+  const handleChange = name => e => {
+    const ratioInput = e.target.value.replace(/,/g, '.');
+    const isNumberValid = /^\d+(?:[.])?\d*$/.test(ratioInput);
+    const nextRatioInput = isNumberValid ? ratioInput : ratioInput.replace(/[^0-9.]/g, '');
+    const formatDecimal = nextRatioInput.match(/^\d+(?:[.])?\d{0,2}/);
+    onRatioChange((formatDecimal && formatDecimal[0]) || nextRatioInput, name);
+    setTooltipPosition(name ?? '');
+  };
+
+  return (
+    <Wrapper>
+      {ratioValues && (
+        <>
+          <BoxPanel width={1 / 2}>
+            <Typography variant="p" textAlign="center" marginBottom="9px">
+              Current
+            </Typography>
+            <List>
+              {ratioValues.map(({ name, percent }) => (
+                <ListItem key={name + percent} hasTitle={!!name}>
+                  {name && <Typography variant="p">{name === 'Loans' ? 'Borrower' : name}</Typography>}
+                  <Typography variant="h2">{percent}%</Typography>
+                </ListItem>
+              ))}
+            </List>
+          </BoxPanel>
+          <BoxPanel width={1 / 2}>
+            <Typography variant="p" textAlign="center" marginBottom="9px">
+              New
+            </Typography>
+            <List>
+              {ratioValues.map(({ name, percent }, index) => (
+                <Tooltip
+                  key={name + percent}
+                  containerStyle={{ width: 'auto' }}
+                  refStyle={{ display: 'block' }}
```

Comment:
What is it for? Is there another solution? I don't like to change many basic components.

Reply:
@octocat512 UI is broken when I wrap the item inside the tooltip.  Please take a look at the image below.
Because `inline-block` does not stretch to fit parent width, I need to modify the CSS of ReferenceElement.
Do you have any solution to fix it instead?
![image](https://user-images.githubusercontent.com/17588370/132944733-1c83733f-c9d5-40a1-94e9-09cd04d21a02.png)
![ev3](https://user-images.githubusercontent.com/17588370/132944775-4ad1888f-8c9b-4164-ab67-9c8d237bdb17.png)